### PR TITLE
mpl2: io blockages refactor / remove member variables from top class

### PR DESCRIPTION
**Pull request by AcKoucher**

### CODE REVIEW

**Comment by AcKoucher** [comment]
Running CI

File: src/mpl2/src/hier_rtlmp.cpp [hunk 1]
```diff
@@ -3045,53 +3032,71 @@ void HierRTLMP::calculateMacroTilings(Cluster* cluster)
...
+  const float depth = computeIOBlockagesDepth(io_spans);
 
-  const std::vector<std::pair<float, float>> tilings
-      = root_cluster_->getMacroTilings();
-  const float ar = root_cluster_->getHeight() / root_cluster_->getWidth();
-  float max_height = std::sqrt(tilings[0].first * tilings[0].second * ar);
-  float max_width = max_height / ar;
-  max_width = ((root_ux - root_lx) - max_width);
-  max_height = ((root_uy - root_ly) - max_height);
+  const Rect root(root_cluster_->getX(),
```
Comment (maliberty) [inline src/mpl2/src/hier_rtlmp.cpp:3044]: Cluster::getX returns a float - is this intended to be rounded to an int?  Likewise the rest.

File: src/mpl2/src/hier_rtlmp.cpp [hunk 2]
```diff
@@ -3045,53 +3032,71 @@ void HierRTLMP::calculateMacroTilings(Cluster* cluster)
...
+  IOSpans io_spans;
 
-  int floorplan_lx = micronToDbu(floorplan_lx_, dbu_);
-  int floorplan_ux = micronToDbu(floorplan_ux_, dbu_);
-  int floorplan_uy = micronToDbu(floorplan_uy_, dbu_);
+  odb::Rect die = block_->getDieArea();
+
+  // Initialize spans based on the dimensions of the die area.
+  io_spans[L] = std::pair<float, float>(dbuToMicron(die.yMax(), dbu_),
+                                        dbuToMicron(die.yMin(), dbu_));
```
Comment (maliberty) [inline src/mpl2/src/hier_rtlmp.cpp:3089]: You can write this as :
io_spans[L] = {dbuToMicron(die.yMax(), dbu_), dbuToMicron(die.yMin(), dbu_)};

File: src/mpl2/src/hier_rtlmp.cpp (hunk 1 above)
Comment (AcKoucher) [reply src/mpl2/src/hier_rtlmp.cpp:3044]: Here Rect is mpl2::Rect which is in micron, not odb::Rect

Comment (maliberty) [reply src/mpl2/src/hier_rtlmp.cpp:3044]: Ok (though we really need to nuke this micron code)

**Comment by maliberty** [comment]
What was the CI result?

**Comment by AcKoucher** [comment]
Secure-CI passed.

**Review by maliberty — APPROVED** [review]

**Pull request merged**